app/src/main/java/io/intrepid/contest/screens/entrysubmission/join/JoinActivity.java

```diff
@@ -68,4 +74,38 @@ public void hideSubmitButton() {
     public void showEntryNameScreen() {
         startActivity(EntryNameActivity.makeIntent(this));
     }
+
+    @Override
+    public void showInvalidCodeErrorMessage() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(R.string.error_redeeming_code)
+                .setTitle(R.string.error_redeeming_code_title)
+                .setNeutralButton(R.string.common_ok, (dialog, id) -> {
+                });
+        builder.create().show();
+    }
+
+    @Override
+    public void showApiErrorMessage() {
+        Context context = getApplicationContext();
+        int duration = Toast.LENGTH_SHORT;
+        Toast toast = Toast.makeText(context, "API endpoint not implemented", duration);
```

Comment:
If this text will be shown to the user, it should be defined as a string resource.